streets_reader/src/split_ways.rs

```diff
@@ -202,24 +204,86 @@ pub fn split_up_roads(
     for (pt, dir) in input.traffic_signals {
         if let Some(r) = pt_to_road.get(&pt) {
             // The road might've crossed the boundary and been clipped
-            if let Some(road) = streets.roads.get(r) {
-                // Example: https://www.openstreetmap.org/node/26734224
-                if road.highway_type != "construction" {
+            if let Some(road) = streets.roads.get_mut(r) {
+                // On a one-way road, specifying direction is redundant, so infer from there too
+                if let Some(dir) = dir.or_else(|| road.oneway_for_driving()) {
+                    // Update the intersection control type
                     let i = if dir == Direction::Fwd {
                         road.dst_i
                     } else {
                         road.src_i
                     };
                     let i = streets.intersections.get_mut(&i).unwrap();
+                    // TODO Maybe we should do this later, as a consequence of TrafficInterruption
+                    // on incoming roads?
                     if !i.is_map_edge() {
                         i.control = IntersectionControl::Signalled;
                     }
+
+                    // Specify the explicit vehicle stop line
+                    if let Some((dist, _)) = road.reference_line.dist_along_of_point(pt.to_pt2d()) {
+                        let stop_line = if dir == Direction::Fwd {
+                            &mut road.stop_line_end
+                        } else {
+                            &mut road.stop_line_start
+                        };
+                        stop_line.vehicle_distance = Some(dist);
+                        stop_line.interruption = TrafficInterruption::Signal;
+                    }
+                    // TODO If dist_along_of_point fails, it's because we smoothed the line. This
```

Comment:
So one of my big questions is this: ultimately I think we want the stop line distance to be based on the untrimmed center, just like `trim_start`. We could do `center_line.project_pt(node_pt)` here, but any part of the code that modifies reference_line or placement will need to do something to preserve the stop line distances.

We could instead store the raw `Pt2D` position of a stop line, and do this `project_pt` thing whenever we need it (just for rendering now, eventually for a traffic simulation rule or for inferring movement geometry or something else). Feels like a messier data model, but maybe if we wrap with a nice API it's OK?

Or maybe when we modify reference line, we need to do so in a kind of "transactional" way -- not directly, but something like `road.update_reference_line(polyline, placement)`. This will first calculate the current `Pt2D` of any stop lines, then change the reference line, then do `project_pt` and calculate the equivalent distance with the new center.

Reply:
I think it would be ideal to have the distance relative to the center line, because that's what we use to practically determine the position of the road later. Reference line probably shouldn't be modified (it's basically just the OSM geom), but the centerline might change. (It would be great to have a transformation that infers missing pacement tags based on lane connectivity, because it is undertagged). I think I like the idea of using a method to update the center line which updates these distances (and the `trim` distances) appropriately.

Reply:
Sounds good, so I'll try that "transactional" approach when I pick this back up